proto/src/proto.rs

```diff
@@ -418,6 +419,33 @@ pub struct LdapMatchingRuleAssertion {
     pub dn_attributes: bool, // DEFAULT FALSE
 }
 
+impl LdapMatchingRuleAssertion {
+    pub fn from_strings(left: String, right: String) -> Self {
+        let match_value = right.to_string();
+        let mut split = left.split(':').collect::<VecDeque<_>>();
```

Comment:
Nitpick: allocation is not required here either, it seems possible to work with the iterator directly.